src/main/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -30,57 +30,63 @@
     private int reviews = 0;
     private long totalProd = 0;
     private long totalUsers = 0;
-    HashMap<String, Long> productsTotal = new HashMap<String, Long>();
-    HashMap<Long, String> productsTotalReverse = new HashMap<Long, String>();
-    HashMap<String, Long> usersTotal = new HashMap<String, Long>();
-    ArrayList<String> products = new ArrayList<String>();
-    ArrayList<String> users = new ArrayList<String>();
-    ArrayList<String> scores = new ArrayList<String>();
+    HashMap<String, Long> productsTotal = new HashMap<>();
+    HashMap<Long, String> productsTotalReverse = new HashMap<>();
+    HashMap<String, Long> usersTotal = new HashMap<>();
+    List<String> products = new ArrayList<>();
+    List<String> users = new ArrayList<>();
+    List<String> scores = new ArrayList<>();
     PrintWriter out = new PrintWriter(new FileWriter("movies.csv"));
 
     public MovieRecommender (String filename) throws IOException{
         this.filename = filename;
         getData();
     }
 
-    public void getData() throws IOException{
-        String line;
-        String idProduct="";
-        String idUser="";
-        String score;
-        InputStream fileStream = new FileInputStream(filename);
-        InputStream gzipStream = new GZIPInputStream(fileStream);
-        Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
-        BufferedReader buffered = new BufferedReader(decoder);
+    public void getData(){
```

Comment:
I see you have different logic in this method: get the info for the product and the review. Also creates the map with all the info. A better practice is to separate the logic into small and more atomic methods. How would you improve this in your code? What do you think is a consequence of having a lot of work in just one place?